### ISSUE TITLE
Maxwell Split Server Roles w/ config.properties Not Issuing CDC to Kafka Topic

### ISSUE THREAD

**Issue by liquidgenius** [zendesk/maxwell#1299]
I am using config.properties file to store the Maxwell metadata on a separate MySQL server from the server where CDC occurs. ([split server roles](http://maxwells-daemon.io/config/#split-server-roles))

**Desired split server setup:**
![split_server_roles](https://user-images.githubusercontent.com/3987909/61831212-0525df80-ae22-11e9-914d-5d64c6aec084.png)

**On Maxwell execution, no errors are thrown, and Maxwell creates its metadata database/schema on the Source DB server, however no JSON is emitted to the Kafka topic when CDC is introduced. A console producer and consumer connected to the same Kafka topic can successfully transfer JSON through Kafka.**

My thought is that the documentation on permissions for the Maxwell user may be different for using Split Server Roles, or something else competely.

For the sake of this issue, Replication Host refers to the MySQL database that Maxwell is watching for CDC occurrences. In the config the relevant fields are:
- replication_host
- replication_user
- replication_password
- replication_port

Metadata Host refers to the MySQL database that stores Maxwell's schema database and related tables, or as defined in the docs; the 'database to store schema and position in'. Our source database schema is simply named 'source.' In the config the relevant fields are:
- host
- user
- password
- port
- schema_database (set to the name of the database being replicated from; 'source')

**The maxwell user for both Replication and Metadata databases is set up the same:**
```
CREATE USER 'md'@'%' IDENTIFIED BY '**********';
GRANT ALL ON source.* TO 'md'@'%';
GRANT SELECT, REPLICATION CLIENT, REPLICATION SLAVE ON *.* TO 'md'@'%';
```

**Replication maxwell user GRANTS:**
```
SHOW GRANTS FOR 'md'@'%';
GRANT SELECT, REPLICATION SLAVE, REPLICATION CLIENT ON *.* TO 'md'@'%'
GRANT ALL PRIVILEGES ON `source`.* TO 'md'@'%'
```

**Metadata maxwell user GRANTS:**
```
SHOW GRANTS FOR 'md'@'%';
GRANT SELECT, REPLICATION SLAVE, REPLICATION CLIENT ON *.* TO 'md'@'%'
GRANT ALL PRIVILEGES ON `source`.* TO 'md'@'%'
```

**The config.properties file is:**
```
# tl;dr config
log_level=debug
producer=kafka

# replication and metadata host (single server role)
#host=13.00.000.000
#user=md
#password=**********
#port=3306

# metadata host (split server role)
host=3.00.00.000
user=md
password=**********
port=3306
schema_database=source

# replication host (split server role)
replication_host=13.00.000.000
replication_user=md
replication_password=**********
replication_port=3306

# kafka
kafka.bootstrap.servers=b-2.obsfucated:9092,b-1.obsfucated:9092,b-3.obsfucated:9092
kafka_topic=testing_%{database}_%{table}
output_dll=true
ddl_kafka_topic=testing_source_ddls
kafka.compression.type=snappy
kafka.retries=0
kafka.acks=1

# filtering
filter= exclude: *.*,  include: source.companies
```

**Maxwell debug output:**
```
Using kafka version: 1.0.0
19:51:38,220 WARN  MaxwellMetrics - Metrics will not be exposed: metricsReportingType not configured.
19:51:38,778 INFO  SchemaStoreSchema - Creating source database
19:51:38,956 INFO  ProducerConfig - ProducerConfig values: 
        acks = 1
        batch.size = 16384
        bootstrap.servers = [b-2.obsfucated:9092, b-1.obsfucated:9092, b-3.obsfucated:9092]
        buffer.memory = 33554432
        client.id = 
        compression.type = snappy
        connections.max.idle.ms = 540000
        enable.idempotence = false
        interceptor.classes = null
        key.serializer = class org.apache.kafka.common.serialization.StringSerializer
        linger.ms = 0
        max.block.ms = 60000
        max.in.flight.requests.per.connection = 5
        max.request.size = 1048576
        metadata.max.age.ms = 300000
        metric.reporters = []
        metrics.num.samples = 2
        metrics.recording.level = INFO
        metrics.sample.window.ms = 30000
        partitioner.class = class org.apache.kafka.clients.producer.internals.DefaultPartitioner
        receive.buffer.bytes = 32768
        reconnect.backoff.max.ms = 1000
        reconnect.backoff.ms = 50
        request.timeout.ms = 30000
        retries = 0
        retry.backoff.ms = 100
        sasl.jaas.config = null
        sasl.kerberos.kinit.cmd = /usr/bin/kinit
        sasl.kerberos.min.time.before.relogin = 60000
        sasl.kerberos.service.name = null
        sasl.kerberos.ticket.renew.jitter = 0.05
        sasl.kerberos.ticket.renew.window.factor = 0.8
        sasl.mechanism = GSSAPI
        security.protocol = PLAINTEXT
        send.buffer.bytes = 131072
        ssl.cipher.suites = null
        ssl.enabled.protocols = [TLSv1.2, TLSv1.1, TLSv1]
        ssl.endpoint.identification.algorithm = null
        ssl.key.password = null
        ssl.keymanager.algorithm = SunX509
        ssl.keystore.location = null
        ssl.keystore.password = null
        ssl.keystore.type = JKS
        ssl.protocol = TLS
        ssl.provider = null
        ssl.secure.random.implementation = null
        ssl.trustmanager.algorithm = PKIX
        ssl.truststore.location = null
        ssl.truststore.password = null
        ssl.truststore.type = JKS
        transaction.timeout.ms = 60000
        transactional.id = null
        value.serializer = class org.apache.kafka.common.serialization.StringSerializer

19:51:38,988 DEBUG Metrics - Added sensor with name bufferpool-wait-time
19:51:38,997 DEBUG Metrics - Added sensor with name buffer-exhausted-records
19:51:39,074 DEBUG Metadata - Updated cluster metadata version 1 to Cluster(id = null, nodes = [b-2.obsfucated:9092 (id: -1 rack: null), b-3.obsfucated:9092 (id: -3 rack: null), b-1.obsfucated:9092 (id: -2 rack: null)], partitions = [])
19:51:39,083 DEBUG Metrics - Added sensor with name produce-throttle-time
19:51:39,093 DEBUG Metrics - Added sensor with name connections-closed:
19:51:39,094 DEBUG Metrics - Added sensor with name connections-created:
19:51:39,094 DEBUG Metrics - Added sensor with name successful-authentication:
19:51:39,095 DEBUG Metrics - Added sensor with name failed-authentication:
19:51:39,096 DEBUG Metrics - Added sensor with name bytes-sent-received:
19:51:39,097 DEBUG Metrics - Added sensor with name bytes-sent:
19:51:39,099 DEBUG Metrics - Added sensor with name bytes-received:
19:51:39,101 DEBUG Metrics - Added sensor with name select-time:
19:51:39,102 DEBUG Metrics - Added sensor with name io-time:
19:51:39,107 DEBUG Metrics - Added sensor with name batch-size
19:51:39,107 DEBUG Metrics - Added sensor with name compression-rate
19:51:39,108 DEBUG Metrics - Added sensor with name queue-time
19:51:39,108 DEBUG Metrics - Added sensor with name request-time
19:51:39,108 DEBUG Metrics - Added sensor with name records-per-request
19:51:39,109 DEBUG Metrics - Added sensor with name record-retries
19:51:39,109 DEBUG Metrics - Added sensor with name errors
19:51:39,110 DEBUG Metrics - Added sensor with name record-size
19:51:39,112 DEBUG Metrics - Added sensor with name batch-split-rate
19:51:39,115 INFO  AppInfoParser - Kafka version : 1.0.0
19:51:39,115 INFO  AppInfoParser - Kafka commitId : aaa7af6d4a11b29d
19:51:39,116 DEBUG KafkaProducer - [Producer clientId=producer-1] Kafka producer started
19:51:39,122 DEBUG Sender - [Producer clientId=producer-1] Starting Kafka producer I/O thread.
19:51:39,242 DEBUG MysqlPositionStore - Writing binlog position to source.positions: Position[BinlogPosition[master.000002:154], lastHeartbeat=0], last heartbeat read: 0
19:51:39,246 INFO  Maxwell - Maxwell v1.22.3 is booting (MaxwellKafkaProducer), starting at Position[BinlogPosition[master.000002:154], lastHeartbeat=0]
19:51:39,434 DEBUG MysqlSavedSchema - looking to restore schema at target position Position[BinlogPosition[master.000002:154], lastHeartbeat=0]
19:51:39,436 INFO  AbstractSchemaStore - Maxwell is capturing initial schema
19:51:39,438 DEBUG MysqlSavedSchema - Capturing schemas...
19:51:39,450 DEBUG MysqlSavedSchema - Starting schema capture of 3 databases...
19:51:39,450 DEBUG MysqlSavedSchema - 1/3 Capturing mysql...
19:51:39,559 DEBUG MysqlSavedSchema - 2/3 Capturing source...
19:51:39,565 DEBUG MysqlSavedSchema - 3/3 Capturing sys...
19:51:39,602 DEBUG MysqlSavedSchema - 3 database schemas captured!
19:51:39,603 DEBUG MysqlSavedSchema - Capturing Default Charset
19:51:39,782 INFO  BinlogConnectorReplicator - Setting initial binlog pos to: master.000002:154
19:51:39,898 INFO  BinaryLogClient - Connected to 13.000.000.000:3306 at master.000002/154 (sid:6379, cid:7)
19:51:39,899 INFO  BinlogConnectorLifecycleListener - Binlog connected.
19:51:41,239 DEBUG MysqlPositionStore - writing heartbeat: 1563997901239 (last heartbeat written: 1563997900230)
19:51:51,245 DEBUG MysqlPositionStore - writing heartbeat: 1563997911245 (last heartbeat written: 1563997901239)
```

**Comment by liquidgenius** [comment]
I don't understand how the privileges work in conjunction with the schema_database parameter. If schema_database is set to something other than 'maxwell' do you set different privileges for the maxwell user?

**Comment by osheroff** [comment]
#1311 merged, closing this.  hope it's fixed!
